pkg/service/service_test.go

```diff
@@ -2737,3 +2740,124 @@ func TestNotifyHealthCheckUpdatesSubscriber(t *testing.T) {
 	// closes the channel.
 	<-cbCh1
 }
+
+func initializeNetns(t *testing.T, ns *netns.NetNS, addr string) net.Conn {
+	var conn net.Conn
+	assert.NoError(t, ns.Do(func() error {
+		ls, err := netlink.LinkList()
+		assert.NoError(t, err)
+		for _, l := range ls {
+			// Netns should be default created with loopback dev
+			// we assign a localhost address to it to allow us to
+			// bind sockets.
+			if l.Attrs().Name == "lo" {
+				netlink.AddrAdd(l, &netlink.Addr{
+					IPNet: &net.IPNet{
+						IP:   net.ParseIP("127.0.0.1"),
+						Mask: net.IPv4Mask(0xff, 0xff, 0xff, 0xff),
+					},
+				})
+			}
+			_, err := netlink.AddrList(l, unix.AF_INET)
+			assert.NoError(t, err)
+		}
+		conn, err = net.Dial("udp", addr)
+		assert.NoError(t, err)
+		conn.Write([]byte("ping"))
+		return err
+	}))
+	return conn
+}
+
+func TestTerminateUDPConnectionsToBackend(t *testing.T) {
+	testutils.PrivilegedTest(t)
+
+	ns1, err := netns.New()
+	require.NoError(t, err)
+	// ns2 has a connection that should not be matched due
+	// to a different port.
+	ns2, err := netns.New()
+	require.NoError(t, err)
+	// ns3 will match revnat, but with a different cookie value
+	// so we expect to avoid a socket close (i.e. this is
+	// the case where we have matching tuple values, but
+	// in an unexpected network namespace).
```

Comment:
" so we expect to avoid a socket close (i.e. this is
	// the case where we have matching tuple values, but
	// in an unexpected network namespace)."

Not sure what you mean by "an unexpected network namespace. As mentioned, we match socket cookie and not netns cookie. 

I would suggest making slight changes in order to make it a realistic unit test case - add connections in host netns and pod netns and check whether the expected connections are closed based on the config `option.Config.BPFSocketLBHostnsOnly`.